### DISCUSSION TITLE
Username and password for AllureDownloadUrl

### DISCUSSION THREAD

**Discussion by vyalekseev** [allure-framework/allure2#2182] [Questions & Support]
I need to download allure-command-line from a local nexus server. But when trying to download it, maven tries to connect without username and password specified in the file settings.xml accordingly, nothing is downloaded without authorization (401 error). How to specify username and password in a file pom.xml or force them to take them from a file settings.xml ?

```
<plugin>
      <groupId>io.qameta.allure</groupId>
      <artifactId>allure-maven</artifactId>
      <version>2.10.0</version>
       <configuration>
            <reportVersion>${allure.version}</reportVersion>
            <allureDownloadUrl>${allure.cmd.url}/io/qameta/allure/allure-commandline/${allure.version}/allure-commandline-${allure.version}.zip</allureDownloadUrl>
         </configuration>    
    </plugin>
```

**Comment by baev** [comment]
Here are two approaches you could take:

Configure your Nexus Repository as one of the Maven repositories and omit the allureDownloadUrl. Without specifying allureDownloadUrl, the Allure Maven plugin will default to using Maven's inherent dependency resolution system.
Establish a Maven Proxy that grants access to the allureDownloadUrl without requiring a password.

**Comment by suponev** [comment]
If your host has basic authentication enabled, you can replace the link with
https://username:password@www.example.com
If your username has "@"
replace it with %40
